src/printer.rs

```diff
@@ -315,17 +308,61 @@ impl<'a> Printer for InteractivePrinter<'a> {
         };
 
         let description = &input.description;
+        let metadata = &input.metadata;
+
+        // We use this iterator to have a deterministic order for
+        // header components. HashSet has arbitrary order, but Vec is ordered.
+        let header_components: Vec<StyleComponent> = [
+            (
+                StyleComponent::HeaderFilename,
+                self.config.style_components.header_filename(),
+            ),
+            (
+                StyleComponent::HeaderFilesize,
+                self.config.style_components.header_filesize(),
+            ),
+        ]
+        .iter()
+        .filter(|(_, is_enabled)| *is_enabled)
+        .map(|(component, _)| *component)
+        .collect();
+
+        header_components.iter().try_for_each(|component| {
+            if self.config.style_components.grid() {
+                write!(
+                    handle,
+                    "{}{}",
+                    " ".repeat(self.panel_width),
+                    self.colors
+                        .grid
+                        .paint(if self.panel_width > 0 { "│ " } else { "" }),
+                )?;
```

Comment:
To make this code easier to understand, can you introduce some helper functions please? I _think_ this block can be moved into a helper function for example.